huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template_group.go

```diff
@@ -186,224 +187,325 @@ func ResourceTranscodingTemplateGroup() *schema.Resource {
+	httpUrl := "v1/{project_id}/template_group/transcodings"
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = fmt.Sprintf("%s?group_id=%s", getPath, groupId[0])
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
 	}
-	templateGroup := templateGroupList[0]
 
-	templateIds := make([]string, len(*templateGroup.TemplateIds))
-	for i, v := range *templateGroup.TemplateIds {
-		templateIds[i] = strconv.FormatInt(int64(v), 10)
+	resp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, err
 	}
 
-	mErr := multierror.Append(nil,
-		d.Set("region", config.GetRegion(d)),
-		d.Set("name", templateGroup.Name),
-		d.Set("audio", flattenAudio(templateGroup.Audio)),
-		d.Set("video_common", flattenVideoCommon(templateGroup.VideoCommon)),
-		d.Set("videos", flattenVideos(templateGroup.Videos)),
-		d.Set("template_ids", templateIds),
-		setCommonAttrs(d, templateGroup.Common),
-	)
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, err
+	}
 
-	if err = mErr.ErrorOrNil(); err != nil {
-		return diag.Errorf("error setting MPC transcoding template group fields: %s", err)
+	templateGgroup := utils.PathSearch("template_group_list|[0]", respBody, nil)
+	if templateGgroup == nil {
+		return nil, golangsdk.ErrDefault404{}
 	}
 
-	return nil
+	return templateGgroup, nil
 }
 
-func resourceTranscodingTemplateGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(*config.Config)
-	client, err := config.HcMpcV1Client(config.GetRegion(d))
-	if err != nil {
-		return diag.Errorf("error creating MPC client : %s", err)
+func flattenTemplateGroupVideos(resp interface{}) []map[string]interface{} {
+	rawArray := resp.([]interface{})
+	if len(rawArray) == 0 {
+		return nil
 	}
 
-	updateOpts := mpc.ModifyTransTemplateGroup{
-		GroupId:     d.Id(),
-		Name:        utils.String(d.Get("name").(string)),
-		Videos:      buildVideosOpts(d.Get("videos").([]interface{})),
-		VideoCommon: buildVideoCommonOpts(d.Get("video_common").([]interface{})),
-		Audio:       buildAudioOpts(d.Get("audio").([]interface{})),
-		Common:      buildCommonOpts(d),
+	rst := make([]map[string]interface{}, len(rawArray))
+	for i, v := range rawArray {
+		params := map[string]interface{}{
+			"width":   utils.PathSearch("width", v, nil),
+			"height":  utils.PathSearch("height", v, nil),
+			"bitrate": utils.PathSearch("bitrate", v, nil),
+		}
+		rst[i] = params
 	}
 
-	updateReq := mpc.UpdateTemplateGroupRequest{
-		Body: &updateOpts,
+	return rst
+}
+
+func flattenTemplateGroupAudio(resp interface{}) []map[string]interface{} {
+	channels := int(utils.PathSearch("channels", resp, float64(0)).(float64))
+	if resp != nil && channels != 0 {
+		return []map[string]interface{}{
+			{
+				"output_policy": utils.PathSearch("output_policy", resp, nil),
```

Comment:
It seems that this field also has a default value copy.
This processing is inconsistent with the original logic.